sklearn/manifold/_isomap.py

```diff
@@ -2,14 +2,18 @@
 
 # Author: Jake Vanderplas  -- <vanderplas@astro.washington.edu>
 # License: BSD 3 clause (C) 2011
+import warnings
 
 import numpy as np
+from scipy.sparse.csgraph import shortest_path
+from scipy.sparse.csgraph import connected_components
+
 from ..base import BaseEstimator, TransformerMixin
 from ..neighbors import NearestNeighbors, kneighbors_graph
 from ..utils.validation import check_is_fitted
-from ..utils.graph import graph_shortest_path
```

Comment:
I know that this what not part of the public API because it was not documented, but we might still want to make a deprecation cycle before removing it. It would have started with a leading underscore, then I would not suggest that.

Reply:
Sounds good, I will add a deprecation cycle.

Do you think it would make sense to remove our implementation, add `from scipy.sparse.csgraph import shortest_path as graph_shortest_path`, and consider all differences to be bugfixes (they are definitely bugfixes)? Or would you rather keep our own implementation during the deprecation cycle?

Reply:
I would consider the sync as a bug fix indeed and it would be completely fine. I think this is a great idea indeed. In the deprecation warning, we could point out the scipy version indeed.